zaza/openstack/charm_tests/saml_mellon/setup.py

```diff
@@ -161,3 +165,88 @@ def attach_saml_resources(application="keystone-saml-mellon"):
         fp.flush()
         zaza.model.attach_resource(
             application, sp_signing_keyinfo_name, fp.name)
+
+
+def _attach_saml_resources_local_idp(keystone_saml_mellon_app_name=None,
+                                     test_saml_idp_app_name=None):
+    """Attach resources to the Keystone SAML Mellon and the local IdP."""
+    test_saml_idp_unit = zaza.model.get_units(
+        test_saml_idp_app_name).pop()
+    action_result = zaza.model.run_action(
+        test_saml_idp_unit.name, 'get-idp-metadata')
+    idp_metadata = action_result.data['results']['output']
+
+    with tempfile.NamedTemporaryFile(mode='w', suffix='.xml') as fp:
+        fp.write(idp_metadata)
+        fp.flush()
+        zaza.model.attach_resource(
+            keystone_saml_mellon_app_name, "idp-metadata", fp.name)
```

Comment:
There's a pattern developing here (this is repeated 3 times).  Would breaking out a helper that does this and attaches the resource (or even add a new helper), help with the clarity and intention of the code.  e.g.

```python
zaza.utils.attach_resource(app, "idp-metadata', idp_metadata, ".xml")
```

Reply:
Good suggestion. Will implement it.